<launch>

  <!-- Arguments -->
  <arg name = "stereo" default = "stereo_down" />
  <arg name="enable_decimate_x1" default="false"/>
  <arg name="enable_decimate_x2" default="false"/>
  <arg name="enable_decimate_x4" default="true"/>
  <arg name="resolution" default="full"/>

  <!-- Bagfiles -->

  <!-- Cala Lliteres -->
  <!-- <arg name = "bagfile" default = "/home/uib/bagfiles/cala_lliteres/2020_07_23_cala_lliteres/15_51_42/images_2020-07-23-16-12-25_2.bag" />  -->
  
  <!-- Valldemossa -->
  <!-- <arg name = "bagfile" default = "/home/uib/bagfiles/valldemossa/2016_09_21_valldemossa/2016_09_21_valldemossa/160921_1659/20160921_165907_0000.bag" /> -->

  <!-- Bendinat good day -->
  <arg name = "bagfile" default = "/home/uib/bagfiles/bendinat/12_24_51/images_2021-01-19-12-31-10_1.bag" />

  <!-- Play the bagfile -->
  <node pkg = "rosbag" type = "play" name = "rosbag" args = "--clock $(arg bagfile)" />

  <!-- Run the stereo image proc -->
  <group>
    <include file="/home/uib/catkin_ws/src/image_pipeline/stereo_image_proc/launch/stereo_proc.launch">
    <arg name="stereo" value="$(arg stereo)" />
    <arg name="resolution" value="$(arg resolution)" />
    <arg name="enable_decimate_x1" value="$(arg enable_decimate_x1)" />
    <arg name="enable_decimate_x2" value="$(arg enable_decimate_x2)" />
    <arg name="enable_decimate_x4" value="$(arg enable_decimate_x4)" />
    </include>
  </group>

  <!-- Run feature tracker SIFT -->
  <node pkg = "viso2_ros" type = "testSIFT2" name = "testSIFT2" output = "screen" if="$(eval enable_decimate_x1 == true)">
    <remap from = "stereo" to = "$(arg stereo)"/>
  </node>

  <node pkg = "viso2_ros" type = "testSIFT2" name = "testSIFT2" output = "screen" if="$(eval enable_decimate_x2 == true)">
    <remap from = "stereo" to = "$(arg stereo)"/>
    <remap from = "/stereo_down/left/image_rect" to = "/stereo_down/scaled_x2/left/image_rect"/>
    <remap from = "/stereo_down/right/image_rect" to = "/stereo_down/scaled_x2/right/image_rect"/> 
  </node>

  <node pkg = "viso2_ros" type = "testSIFT2" name = "testSIFT2" output = "screen" if="$(eval enable_decimate_x4 == true)">
    <remap from = "stereo" to = "$(arg stereo)"/>
    <remap from = "/stereo_down/left/image_rect" to = "/stereo_down/scaled_x4/left/image_rect"/>
    <remap from = "/stereo_down/right/image_rect" to = "/stereo_down/scaled_x4/right/image_rect"/> 
  </node>
  
</launch>